CMakeLists.txt

```diff
@@ -73,21 +74,79 @@ if(COVERAGE)
        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -coverage")
 endif()
 
-option(EXPERIMENTAL "use experimental features" OFF)
-if(EXPERIMENTAL)
-    add_definitions(-DEXPERIMENTAL)
-endif(EXPERIMENTAL)
-
-set(SOURCE_FILES src/libpmemkv.cc src/libpmemkv.h
-    src/engines/blackhole.h src/engines/blackhole.cc
-    src/engines/vsmap.h src/engines/vsmap.cc
-    src/engines/vcmap.h src/engines/vcmap.cc
-    src/engines/cmap.h src/engines/cmap.cc
-    src/engines-experimental/tree3.h src/engines-experimental/tree3.cc
-    src/engines-experimental/stree.h src/engines-experimental/stree.cc
-    src/engines-experimental/stree/persistent_b_tree.h src/engines-experimental/stree/pstring.h
-    src/engines-experimental/caching.h src/engines-experimental/caching.cc
+# Each engine is switched on separately
+# By default all experimental engines are turned off
+option(ENABLE_VSMAP_ENGINE "enable installation and testing of VSMap engine" ON)
+option(ENABLE_VCMAP_ENGINE "enable installation and testing of VCMap engine" ON)
+option(ENABLE_CMAP_ENGINE "enable installation and testing of CMap engine" ON)
+option(ENABLE_CACHING_ENGINE "enable installation and testing of experimental CachingEngine" OFF)
+option(ENABLE_STREE_ENGINE "enable installation and testing of experimental STree engine" OFF)
+option(ENABLE_TREE3_ENGINE "enable installation and testing of experimental Tree engine" OFF)
```

Comment:
IMHO both names and descriptions are too verbose. 
ENGINE_${name} or ${name}_ENGINE would be enough. ENGINE_${name} is a tiny bit better for automated processing (for example easier to grep/match).
There's no point in saying that option is for "installation and testing" - it should be obvious enough with just "enable [experimental] XXX engine".